### PR TITLE
editoast: add roles check to `views::search`

### PR DESCRIPTION
**Pull request by hamz2a**
Part of #8446

### CODE REVIEW

File: editoast/src/views/search.rs [hunk 1]
```diff
@@ -333,10 +337,19 @@ struct SearchDBResult {
...
     Json(SearchPayload { object, query, dry }): Json<SearchPayload>,
 ) -> Result<impl IntoResponse> {
+    let authorized = authorizer
+        .check_roles([BuiltinRole::InfraRead].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
```
Comment (leovalais) [inline editoast/src/views/search.rs]: It's more complicated than that, the search engine is dynamic and isn't restricted to infra object search. Use `object` to determine the correct roles to check.

**Review by leovalais — CHANGES_REQUESTED** [review]

File: editoast/src/views/search.rs [hunk 2]
```diff
@@ -333,10 +338,31 @@ struct SearchDBResult {
...
+    match object.as_str() {
+        "track" | "operationalpoint" | "signal" => roles_to_check.insert(BuiltinRole::InfraRead),
+        "project" | "study" | "scenario" => roles_to_check.insert(BuiltinRole::OpsRead),
+        &_ => {
+            return Err(SearchApiError::ObjectType {
+                object_type: object.to_owned(),
+            }
+            .into())
+        }
+    };
```
Comment (leovalais) [inline editoast/src/views/search.rs]: I think you can write directly:

```suggestion
    let roles: &[BuiltinRole] = match object.as_str() {
        "track" | "operationalpoint" | "signal" => &[BuiltinRole::InfraRead],
        "project" | "study" | "scenario" => &[BuiltinRole::OpsRead],
        _ => {
            return Err(SearchApiError::ObjectType {
                object_type: object.to_owned(),
            }
            .into())
        }
    };
```

**Review by leovalais — APPROVED** [review]
A minor change to avoid mutability, otherwise lgtm

File: editoast/src/views/search.rs (hunk 1 above)
Comment (hamz2a) [reply editoast/src/views/search.rs]: as discussed, we will use compile-time defined strings to check roles based on the `object` field in the `SearchPayload`.

**Pull request merged**